pkg/security/ldapclient/client.go

```diff
@@ -12,7 +12,7 @@ import (
 )
 
 // NewLDAPClientConfig returns a new LDAP client config
-func NewLDAPClientConfig(URL, bindDN, bindPassword, CA string, insecure bool) (Config, error) {
+func NewLDAPClientConfig(URL, bindDN, bindPassword, CA string, insecure bool) (*ldapClientConfig, error) {
```

Comment:
It is idiomatic (["accept interfaces, return struct"](https://en.m.wikipedia.org/wiki/Robustness_principle)). But I remember a linter message claiming that returning unexposed types sucks.
Does it make a difference if the struct is exported or not? All the properties are private.

Reply:
This is a controversial topic, even @sttts commented way back in https://github.com/golang/lint/issues/210 (see https://github.com/golang/lint/issues/210#issuecomment-339244863). The biggest complaint is that the type is not embeddable however an interface implied from it is.

Reply:
I think it is better than before.

Reply:
Alternatively you can do:
```go
NewLDAPClientConfig(...) (Config, error) {
    return NewLDAPClientConfigWithTLSConnectionVerification(..., nil)
}
```

Reply:
I am fine with whatever proposal, i am good with a separate constructor modulo the bigger diff. but since admittedly returning private types is not something common in the k8s code base i think @stlaz's proposal does not look alien to readers of existing code.

Reply:
fixed with the additional constructor.